src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs

```diff
@@ -436,7 +436,7 @@ private static void makeNtlm2Hash(string domain, string userName, ReadOnlySpan<c
                 using (var hmac = IncrementalHash.CreateHMAC(HashAlgorithmName.MD5, pwHash))
                 {
                     // strangely, user is upper case, domain is not.
-                    byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpper(), domain));
+                    byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpperInvariant(), domain));
```

Comment:
@wfurt, I assume this is correct?  I looked at the spec but it doesn't define "Uppercase(...)".

Reply:
I'm not intimately familiar with NTLM, but in general, _ToUpperInvariant_ is the correct thing to do on Windows as far as usernames and other security identifiers are concerned.

Reply:
Would that work for international characters? I just did test on my local Windows 10 and I _can_ create user with localized name e.g. this does not count as special characters. I'm not sure how common that is. 
cc: @filipnavara  for visibility.

Reply:
It's underspecified but I agree with @GrabYourPitchforks's assessment. `ToUpperInvariant` still works for national characters albeit with different rules and it makes sense that authentication would not depend on a specific locale.